std/math/fp2.asm

```diff
@@ -90,6 +95,35 @@ let<T> unpack_ext: Fp2<T> -> (T, T) = |a| match a {
     Fp2::Fp2(a0, a1) => (a0, a1)
 };
 
+/// Whether we need to operate on the F_{p^2} extension field (because the current field is too small).
+let needs_extension: -> bool = || match known_field() {
+    Option::Some(KnownField::Goldilocks) => true,
+    Option::Some(KnownField::BN254) => false,
+    None => panic("The permutation/lookup argument is not implemented for the current field!")
+};
+
+/// Matches whether the correct length is given to operate on the extension field
+let is_extension: int -> bool = |length| match length {
+        1 => false,
+        2 => true,
+        _ => panic("Expected 1 or 2 accumulator columns!")
+};
+
+/// If we don't need an extension field, we can simply set the second component to 0
+let fp2_from_array = |arr, is_ext| if is_ext { Fp2::Fp2(arr[0], arr[1]) } else { from_base(arr[0]) };
+
+/// Asserts wheter we need to operate on the extension field
+let assert_extension = |is_ext| if !is_ext{
```

Comment:
In my opinion, this function does not really look useful to be in isolation. There are a lot of assumptions about how it's used, so maybe extract a bit more of the code around it?